website/docs/schema-generator/execution/data-fetching-environment.md

```diff
@@ -44,4 +44,30 @@ Then the following query would return `"The parentField was foo and the childFie
 ```
 
 You can also use this to retrieve arguments and query information from higher up the query chain. You can see a working
-example in the `graphql-kotlin-spring-example` module [[link](https://github.com/ExpediaGroup/graphql-kotlin/blob/master/examples/spring/src/main/kotlin/com/expediagroup/graphql/examples/query/EnvironmentQuery.kt)].
+example in the `graphql-kotlin-spring-example` module [[link](https://github.com/ExpediaGroup/graphql-kotlin/blob/master/examples/server/spring-server/src/main/kotlin/com/expediagroup/graphql/examples/server/spring/query/EnvironmentQuery.kt)].
+
+```kotlin
+class ProductQueryService : Query {
+
+  fun products(environment: DataFetchingEnvironment): Product {
+      environment.selectionSet.fields.forEach { println("field: ${it.name}") }
+
+    return Product(1, "Product title", 100)
+  }
+}
+
+```
+
+```graphql
+{
+  product {
+    id
+    title
+    price
+  }
+}
+```
+
+You can also use `selectionSet` to retrieve sub field information a client wants. It can be useful to know
```

Comment:
```suggestion
You can also use `selectionSet` to access the selected fields of the current field. It can be useful to know which selections have been requested so the data fetcher can optimize the data access queries. For example, in an SQL-backed system, the data fetcher can access the database and use the field selection criteria to specifically retrieve only the columns that have been requested by the client.
```